app/services/email_checker.rb

```diff
@@ -47,7 +47,7 @@ def parse_address(addr)
   end
 
   def domain_dot_error?
-    domain&.start_with?('.')
+    domain&.start_with?('.') || domain&.match(/\.{2,}/)
```

Comment:
I'm wondering if https://github.com/ministryofjustice/prison-visits-public/blob/005019c3b4ed2207bc4769bb2b88a9cb161e933e/app/services/email_checker.rb#L43 
wouldn't do the trick.
Alternatively It's possible that the URI.parse  or addressable gem https://github.com/sporkmonger/addressable (more reliable than ruby's URI parser)  would be sufficient and more reliable rather than a custom regexp

Reply:
The #parse_address method doesn't produce an error if a user has entered an incorrect email format, specifically if they've added two consecutive dots in the domain.  

I've tried out the addressable gem and it also doesn't pick up on this issue of having two consecutive dots in the domain.  I did research existing gems to deal with this issue, but unfortunately wasn't able to find anything that dealt with this; which is why the custom regexp was implemented.